src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
+	roomId: string,
 ) {
-	socket.emit(emitterName, message, roomId, userName)
+	socket.emit(emitterName, message, userName, timeStamp, roomId)
+}
+
+export function useAllUsersPointsListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<AllUsersPointsRes>()
+	useEffect(() => {
+		function onListenerRes(
+			message: ListenerRes[],
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
+		) {
+			setListenerRes({ message, userName, timeStamp })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+	return listenerRes
 }
 
 export function useSocketListener(listenerName: string) {
-	const [listenerRes, setListenerRes] = useState<ListenerRes>({
-		messageServer: '',
-		roomIdServer: '',
-		userNameServer: '',
-	})
+	const [listenerRes, setListenerRes] = useState<ListenerRes>()
+
 	useEffect(() => {
 		function onListenerRes(
-			messageServer: string,
-			roomIdServer: string,
-			userNameServer: string,
+			message: string,
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
 		) {
-			setListenerRes({ messageServer, roomIdServer, userNameServer })
+			setListenerRes({ message, userName, timeStamp })
+		}
+		socket.on(listenerName, onListenerRes)
```

Comment:
todo: In this file you referred to socket event names (eg 'join-room'), in 2 different ways: `listenerName` and `emitterName`. But in reality the best name is `eventName` (based on [socket docs](https://socket.io/docs/v4/client-api/#socketoneventname-callback)). Naming consistency is essential to understand the data flow quicker.

Reply:
I honestly prefer 2 separate names and chose to ignore the socket.io docs except on the server. It is a listenerName in one place, an emitterName in another place. Naming them eventName for 2 different things doesn't make sense. On the server, since it is literally a listen and an emit in the exact same function, that's different, so it is eventName. 

I'll think about changing it. But I prefer it as is.

Reply:
But a `listenerName` and a `emitterName` refer to the same thing: `join-room`, `story-points`, etc. If you want, leave it as it is. Hopefully in the future, you'll understand better what I mean :)

Reply:
I'm trying eventName. If I find it confusing, I can change it back. But I'm sure I won't. I'll trust you. 😉

Reply:
DONE